src/util/HeaderUtil.ts

```diff
@@ -106,9 +106,20 @@ export interface AcceptDatetime extends AcceptHeader { }
  * Contents of a HTTP Content-Type Header.
  * Optional parameters Record is included.
  */
-export interface ContentType {
-  value: string;
-  parameters: Record<string, string>;
+export class ContentType {
+  public constructor(public value: string, public parameters: Record<string, string> = {}) {}
+
+  /**
+   * Serialize this ContentType object to a ContentType header appropriate value string.
+   * @returns The value string, including parameters, if present.
+   */
+  public toHeaderValueString(): string {
+    let str = this.value;
+    if (Object.keys(this.parameters).length > 0) {
+      str += Object.entries(this.parameters).reduce((acc, entry): string => `${acc}; ${entry[0]}=${entry[1]}`, '');
+    }
```

Comment:
I'm thinking we might want to sort these to make sure we have consistent output which makes testing easier. The if-statement is also not really necessary and you could use `str` as starting `acc` value instead of `''`.